### PR TITLE
Add modules-build and create-package tasks

### PR DESCRIPTION
**Pull request by jgoizueta**
# Description

Shortcut

- Story: https://app.shortcut.com/cartoteam/story/259878
- Autolink: [sc-259878]

- Story: https://app.shortcut.com/cartoteam/story/259879
- Autolink: [sc-259879]

This add takss for databricks to generate the SQL script from source files and creats the installation package (to complete [sc-259879](https://app.shortcut.com/cartoteam/story/259879) we need to add a LICENSE to the package in the advanced repo)

The published version of the SQL modules file contains a `@@DATABASE@@` placeholder, like in Snowflake, that must be replaced by the target Databricks schema name before installation.

### CODE REVIEW

File: clouds/databricks/modules/Makefile [hunk 1]
```diff
@@ -34,16 +34,47 @@ test: check check-extra venv
...
+
+build: venv3 clean
+	echo "Building modules..."
+ifndef DB_SCHEMA
+	echo "- Databricks schema not defined, using default"
+	$(eval DB_SCHEMA := default)
+endif
+	$(VENV3_BIN)/python  $(COMMON_DIR)/python_utils/create_modules_sql.py $(BUILD_DIR)/modules.sql $(DB_SCHEMA)
+
+upload-sql:
```
Comment (Jesus89) [inline clouds/databricks/modules/Makefile:59]: upload-sql: build

File: clouds/databricks/Makefile [hunk 2]
```diff
@@ -88,3 +97,13 @@ ci-release-core:
 
 create-headers:
 	$(MAKE) -C libraries/scala create-headers
+
+create-package: clean build
```
Comment (Jesus89) [inline clouds/databricks/Makefile:101]: Needs to be a production build. Why clean is added here?

Suggested

```
create-package:
        $(MAKE) build production=1
        ...
```

File: clouds/databricks/Makefile [hunk 3]
```diff
@@ -88,3 +97,13 @@ ci-release-core:
...
+	echo "- Creating package"
+	rm -rf $(DIST_DIR)
+	mkdir -p $(DIST_DIR)/$(PACKAGE_NAME)
+	cp $(BUILD_DIR)/modules.sql $(DIST_DIR)/$(PACKAGE_NAME)/
+	echo '{"latest_version": "$(PACKAGE_VERSION)"}' > $(DIST_DIR)/metadata.json
+	cd $(DIST_DIR) && zip -r $(PACKAGE_NAME).zip . && cd ..
+	cp $(BUILD_DIR)/modules.sql $(DIST_DIR)/modules-$(PACKAGE_VERSION).sql
```
Comment (Jesus89) [inline clouds/databricks/Makefile:108]: Why is this required `modules-$(PACKAGE_VERSION).sql`?

File: clouds/databricks/Makefile [hunk 4]
```diff
@@ -88,3 +97,13 @@ ci-release-core:
 
 create-headers:
 	$(MAKE) -C libraries/scala create-headers
+
+create-package: clean build
+	echo "- Creating package"
+	rm -rf $(DIST_DIR)
+	mkdir -p $(DIST_DIR)/$(PACKAGE_NAME)
+	cp $(BUILD_DIR)/modules.sql $(DIST_DIR)/$(PACKAGE_NAME)/
```
Comment (Jesus89) [inline clouds/databricks/Makefile:105]: It must include also the JAR file(s) To be able to install everything from the zip

**Review by Jesus89 — CHANGES_REQUESTED** [review]
Several comments were added.

We need to test this also from AT to make sure everything works there too.

File: clouds/databricks/Makefile (hunk 2 above)
Comment (jgoizueta) [reply clouds/databricks/Makefile:101]: 🤔 I thought I had put the production in the CI call (since this is intended to be used from CI), but yes, better add it here. The clean was probably from avoiding old files in local tests, removing it

Comment (Jesus89) [reply clouds/databricks/Makefile:101]: This is what we are doing in other clouds: no clean and build with production=1 to avoid generating packages with custom prefix

File: clouds/databricks/Makefile (hunk 3 above)
Comment (jgoizueta) [reply clouds/databricks/Makefile]: removed now

File: clouds/databricks/common/python_utils/create_drop_functions.py [hunk 5]
```diff
@@ -1,13 +1,22 @@
...
+schema = sys.argv[3] if len(sys.argv) > 3 else None
 
 
 def get_uninstall_for_module(module_path, queries_list):
-    database = os.getenv('DB_SCHEMA')
+    database = schema
     database = 'default' if database is None else database
     _, _, functions = next(os.walk(module_path))
+    # FIXME: should we read the SQL file, extract the @@DB_SCHEMA@@.function_name
+    # frrom the definition and substitute the schema?
```
Comment (vdelacruzb) [inline clouds/databricks/common/python_utils/create_drop_functions.py:19]: type in frrom

File: clouds/databricks/modules/Makefile [hunk 6]
```diff
@@ -34,16 +61,20 @@ test: check check-extra venv
...
 	deactivate
 
-remove: check
+remove: venv3 check
 	echo "Removing modules..."
-	python3 $(COMMON_DIR)/python_utils/create_drop_functions.py
+	$(VENV3_BIN)/python  $(COMMON_DIR)/python_utils/create_drop_functions.py $(SQL_DIR) $(BUILD_DIR)/drop.sql $(DB_SCHEMA)
 	databricks workspace mkdirs $(SQL_DEPLOY_PATH)
-	databricks workspace import --overwrite --language SQL $(SQL_PATH) $(SQL_DEPLOY_PATH)/dropUDF
+	databricks workspace import --overwrite --language SQL $(BUILD_DIR)/drop.sql $(SQL_DEPLOY_PATH)/drop.sql
```
Comment (vdelacruzb) [inline clouds/databricks/modules/Makefile:68]: we can maybe call the output `DROP_FUNCTIONS.sql` as in other providers.

File: clouds/databricks/common/python_utils/create_drop_functions.py (hunk 5 above)
Comment (vdelacruzb) [inline clouds/databricks/common/python_utils/create_drop_functions.py:19]: Actually if you take a look to `DROP_FUNCTIONS.sql` in other providers it drops all the functions installed by schema. It doesn't care about the sql files that we have in the repo and doesn't read them (BTW there is one example of extracting UDFS and procedures in core/clouds/snowflake/common/build_share.js). This was done to avoid keeping private functions that has no named files and public functions that are renamed.

File: clouds/databricks/modules/Makefile (hunk 6 above)
Comment (jgoizueta) [reply clouds/databricks/modules/Makefile:69]: Yes, @Jesus89 also told me about it, we decided to do it in a separate ticket (my initial attempt to list functions using information_schema failed, we'll need to figure out how to do it)

File: clouds/databricks/common/python_utils/create_drop_functions.py (hunk 5 above)
Comment (jgoizueta) [reply clouds/databricks/common/python_utils/create_drop_functions.py:20]: Yep, you're right, we'll do it in a separte ticket

File: clouds/databricks/modules/Makefile (hunk 6 above)
Comment (Jesus89) [reply clouds/databricks/modules/Makefile:69]: Yes, let's put that aside from now.

**Review by Jesus89 — APPROVED** [review]
LGTM!
